src/dunit/attributes.d

```diff
@@ -16,6 +18,40 @@ struct Tag
     string name;
 }
 
+struct EnabledIf
+{
+    bool condition;
+    string reason;
+}
+
+struct DisabledIf
+{
+    bool condition;
+    string reason;
+}
+
+struct EnabledIfEnvironmentVariable
+{
+    string named;
+    string matches = ".*";
+}
+
+struct DisabledIfEnvironmentVariable
+{
+    string named;
+    string matches = ".*";
+}
+
+struct EnabledOnOs
+{
+    OS[] os;
```

Comment:
Consider adding a constructor:

    this(OS[] os...)
    {
        this.os = os;
    }

Then, you can omit the brackets:

    @EnabledOnOs(OS.linux, OS.osx)

Reply:
fixed